docs/current/guides/snippets/build-test-publish-php/dagger.php

```diff
@@ -0,0 +1,277 @@
+                        id
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    QUERY;
+    $runtime = $this->executeQuery($runtimeQuery);
+
+    // add application source code
+    // set file permissions
+    // set environment variables
+    $appQuery = <<<QUERY
+    query {
+      container (id: "$runtime") {
+        withMountedDirectory(path: "/mnt", source: "$sourceDir") {
+          withWorkdir(path: "/mnt") {
+            withExec(args: ["cp", "-R", ".", "/var/www"]) {
+              withExec(args: ["chown", "-R", "www-data:www-data", "/var/www"]) {
+                withExec(args: ["chmod", "-R", "777", "/var/www/storage"]) {
+                  withExec(args: ["chmod", "+x", "/var/www/docker-entrypoint.sh"]) {
+                    id
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    QUERY;
+    $app = $this->executeQuery($appQuery);
+
+    // install Composer
+    // add application dependencies
+    $appWithDepsQuery = <<<QUERY
+    query {
+      container (id: "$app") {
+        withExec(args: ["sh", "-c", "curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer"]) {
+          withWorkdir(path: "/var/www") {
+            withExec(args: ["composer", "install"]) {
+              id
+            }
+          }
+        }
+      }
+    }
+    QUERY;
+    $appWithDeps = $this->executeQuery($appWithDepsQuery);
+    return $appWithDeps;
+  }
+
+  // build image for testing
+  public function buildTestImage() {
+    // build base image
+    $image = $this->buildBaseImage();
+
+    // set test-specific variables
+    $appTestQuery = <<<QUERY
+    query {
+      container (id: "$image") {
+        withEnvVariable(name: "APP_DEBUG", value: "true") {
+          withEnvVariable(name: "LOG_LEVEL", value: "debug") {
+            id
+          }
+        }
+      }
+    }
+    QUERY;
+    $appTest = $this->executeQuery($appTestQuery);
+    return $appTest;
+  }
+
+  // build image for production
+  public function buildProductionImage() {
+    // build base image
+    $image = $this->buildBaseImage();
+
+    // set production-specific variables
+    $appProductionQuery = <<<QUERY
+    query {
+      container (id: "$image") {
+        withEnvVariable(name: "APP_DEBUG", value: "false") {
+          withEnvVariable(name: "APP_NAME", value: "Laravel with Dagger") {
+            withEntrypoint(args: "/var/www/docker-entrypoint.sh") {
+              id
+            }
+          }
+        }
+      }
+    }
+    QUERY;
+    $appProduction = $this->executeQuery($appProductionQuery);
+    return $appProduction;
+  }
+
+  // run unit tests
+  public function testImage($image) {
+    // create database service container
+    $dbQuery = <<<QUERY
+    query {
+      container {
+        from(address: "mariadb:10.11.2") {
```

Comment:
I would do the same thing with the MariaDB version: https://github.com/dagger/dagger/pull/4913/files#r1166893225

Reply:
Done